tests/integration/targets/setup_postgresql_replication/tasks/setup_postgresql_cluster.yml

```diff
@@ -40,28 +24,56 @@
   shell: wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | sudo apt-key add -
   when: ansible_facts.distribution == 'Ubuntu' and ansible_facts.distribution_major_version >= '20'
 
+- name: Add a repository
+  shell: wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | sudo tee /etc/apt/trusted.gpg.d/pgdg.asc &>/dev/null
+  when: ansible_facts.distribution == 'Ubuntu' and ansible_facts.distribution_major_version >= '20'
+
 - name: Add a repository
   shell: apt -y update
   when: ansible_facts.distribution == 'Ubuntu' and ansible_facts.distribution_major_version >= '20'
```

Comment:
It won't let me select them all, but I'd replace lines 19-33 with:
```
- name: Configure Ubuntu 20 for PostgreSQL
  when:
    - ansible_facts['distribution'] == 'Ubuntu'
    - ansible_facts['distribution_major_version'] is version('20', 'ge')
  block:    
    - name: Install wget
      package:
        name: wget
    - name: Add PGDG repository
      lineinfile:
        create: true
        line: 'deb http://apt.postgresql.org/pub/repos/apt $(lsb_release -cs)-pgdg main'
        path: '/etc/apt/sources.list.d/pgdg.list'
        state: 'present'
    - name: Add PGDG GPG key
      get_url:
        dest: '/etc/apt/trusted.gpg.d/ACCC4CF8.asc'
        url: 'https://www.postgresql.org/media/keys/ACCC4CF8.asc'
    - name: Update apt cache
      apt:
        update_cache: true
```

Reply:
@hunleyd your suggestion LGTM, how about merging and backporting this as is and you could submit a quick PR after that (made even via GH WebUI edit button)?

Reply:
pushed to your branch via the UI. also fixed the 'why does the service start automatically' :)